cooltools/api/coverage.py

```diff
@@ -102,24 +125,40 @@ def coverage(
         raise ValueError(
             "Please, specify ignore_diags and/or IC balance this cooler! Cannot access the value used in IC balancing. "
         )
-
+    if clr_weight_name is not None:
+        try:
+            _ = is_cooler_balanced(clr, clr_weight_name, raise_errors=True)
+        except Exception as e:
+            raise ValueError(
+                f"provided cooler is not balanced or {clr_weight_name} is missing"
+            ) from e
+            
     chunks = cooler.tools.split(clr, chunksize=chunksize, map=map, use_lock=use_lock)
 
     if ignore_diags:
         chunks = chunks.pipe(_zero_diags, n_diags=ignore_diags)
 
```

Comment:
here we'd add the balancing transformation step to the `split-apply-combine` like so:
```python
if clr_weight_name is not None:
    pixel_weight_key = "balanced"
    chunks = chunks.pipe(_apply_balancing, bias=bias, balanced_column_name=pixel_weight_key')
else:
    pixel_weight_key = "count"

```